packages/govuk-frontend/src/govuk/component.mjs

```diff
@@ -0,0 +1,31 @@
+import { GOVUKFrontendSupportError } from './errors/index.mjs'
+
+/**
+ * Base Component class
+ *
+ * Centralises the behaviours shared by our components
+ *
+ * @internal
+ * @abstract
+ */
+export class Component {
+  /**
+   * Constructs a new component, validating that GOV.UK Frontend is supported
+   *
+   * @internal
+   */
+  constructor () {
+    this.validateGOVUKFrontendSupport()
```

Comment:
Probably pair a shorter `checkSupport()` with an `isSupported` getter?

Based on our current method names:

```
checkMode()
checkIfAllSectionsOpen()
checkForSessionStorage()
```

Reply:
Just seen your comment, holding off!

Reply:
Sounds fine with me, I'll update tomorrow.

Regarding `isSupported()`, I'd be keen to have it as a utility function in `govuk/common/index.mjs` rather than on the class itself as there's no `this` involved in its execution.